cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandlerInternal.java

```diff
@@ -200,4 +219,72 @@ public void getAppDetailForVersion(HttpRequest request, HttpResponder responder,
         : applicationLifecycleService.getAppDetail(appId);
     responder.sendJson(HttpResponseStatus.OK, GSON.toJson(appDetail));
   }
+
+  /**
+   * Deploy an application.
```

Comment:
Please mention why we have to deploy methods one inetrnally one externally

Reply:
Updated the javadoc comment and mentioned why this internal API is needed.